src/Plugin/Block/AlertBannerBlock.php

```diff
@@ -192,8 +204,15 @@ protected function getCurrentAlertBanners() {
 
     // Load alert banners and add all.
     // Visibility check happens in build, so we get cache contexts on all.
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
     foreach ($published_alert_banners as $alert_banner_id) {
       $alert_banner = $this->entityTypeManager->getStorage('localgov_alert_banner')->load($alert_banner_id);
+
+      // Get alert banner translation if it exists.
+      if ($alert_banner->hasTranslation($langcode)) {
+        $alert_banner = $alert_banner->getTranslation($langcode);
```

Comment:
Shouldn't this be EntityRepository::getTranslationFromContext()

and I assume this has knock on with the fact that this is called from the constructor (and probably shouldn't be).

Reply:
Yeah I was looking into that, I think we might need to get #321 merged first to check this (remove call from constructer).

That said, the alert banner loaded is the correct translation until the condition fields display paths are set to something different for the translation, so I assume there is already some translation context aware loading going on.

I added a test case for the scenarios I was seeing the break on and the above method does work. Is there a particular reason to favour `EntityRepository::getTranslationFromContext()` as opposed to `$alert_banner->getTranslation($langcode)` method?

Reply:
> I added a test case for the scenarios I was seeing the break on and the above method does work. Is there a particular reason to favour EntityRepository::getTranslationFromContext() as opposed to $alert_banner->getTranslation($langcode) method?

Because the former is intended for getting translations to edit them, the later is intended for getting translations to display them in context should they exist. The later does the work of checking if there is a translation for you, checks cachability, and it uses the language fallback system (as configured for the particular site) to decide which language to use.